spec/factories/tailoring.rb

```diff
@@ -32,16 +32,18 @@
       rules do
         result = []
 
-        loop do
-          # sample of rules under profile
-          profile_rules = profile.rules.sample(profile.rules.count / 2)
-          # sample of rules under security guide, outside of the profile
-          sec_guide_rules_avail = security_guide.rules.reject { |rule| profile_rules.include?(rule) }
-          sec_guide_rules = sec_guide_rules_avail.sample(sec_guide_rules_avail.count / 2)
-
-          result = sec_guide_rules + profile_rules
-          # Make sure that the randomized result doesn't match the profile tailoring
-          break if result.to_set(&:id) != profile.rules.to_set(&:id)
+        if profile.rules.any?
```

Comment:
Instead of trying to hack around the factory, maybe it could be called properly at its callsite.

Reply:
Here, I'd actually prefer to keep the condition in the place, altough the callsite has been corrected.

In future, someone can still try to generate tailoring while having no rules under profile, and the condition is a clear hint for why the loop shouldn't be executed.

Reply:
If the thing will not fail under the wrong circumstances, people will not see that what they are doing is wrong...